### PR TITLE
Fix usefulness check for never type

### PR DESCRIPTION
**Pull request by MikailBag**

### CODE REVIEW

File: crates/ra_hir_ty/src/_match.rs [hunk 1]
```diff
@@ -573,15 +573,22 @@ pub(crate) fn is_useful(
...
+    // In those cases, no match arm is useful.
     if let Ty::Apply(ApplicationTy { ctor: TypeCtor::Adt(AdtId::EnumId(enum_id)), .. }) =
         cx.infer[cx.match_expr].strip_references()
     {
         if cx.db.enum_data(*enum_id).variants.is_empty() {
             return Ok(Usefulness::NotUseful);
         }
     }
+    if let Ty::Apply(ApplicationTy { ctor: TypeCtor::Never, .. }) =
+        cx.infer[cx.match_expr].strip_references()
```
Comment (flodiebold) [inline crates/ra_hir_ty/src/_match.rs:588]: maybe a `match` instead of two `if let`s?

Comment (MikailBag) [reply crates/ra_hir_ty/src/_match.rs]: I've changed to match. I can merge branches even further, but imho code will be too comples.

**Comment by flodiebold** [comment]
Thanks!

**Pull request merged**